__integrations__/models/user_data_processing.test.ts

```diff
@@ -140,4 +140,104 @@ describe("Models |> UserDataProcessing", () => {
 
     context.dispose();
   });
+
+  it("should save reason only for FAILED docs", async () => {
+    const context = await createContext(USER_DATA_PROCESSING_MODEL_PK_FIELD);
+    await context.init();
+    const model = new UserDataProcessingModel(context.container);
+
+    const newDoc = {
+      kind: "INewUserDataProcessing" as const,
+      ...aUserDataProcessing,
+      status: UserDataProcessingStatusEnum.WIP,
+      reason: "should not be saved" as NonEmptyString
+    };
+
+    // create a new document
+    const created = await model
+      .create(newDoc)
+      .fold(
+        _ => fail(`Failed to create doc, error: ${toString(_)}`),
+        result => {
+          expect(result).toEqual(
+            expect.objectContaining({
+              ...aUserDataProcessing,
+              status: UserDataProcessingStatusEnum.WIP,
+              version: 0
+            })
+          );
+          expect(result.reason).not.toBeDefined();
+          return result;
+        }
+      )
+      .run();
+
+    // update document
+    const updates = {
+      status: UserDataProcessingStatusEnum.ABORTED,
+      reason: "should not be saved" as NonEmptyString
+    };
+    await model
+      .update({ ...created, ...updates })
+      .fold(
+        _ => fail(`Failed to update doc, error: ${toString(_)}`),
+        result => {
+          expect(result).toEqual(
+            expect.objectContaining({
+              ...aUserDataProcessing,
+              status: UserDataProcessingStatusEnum.ABORTED
+            })
+          );
+          expect(result.reason).not.toBeDefined();
+        }
+      )
+      .run();
+
+    // update document with FAILED
+    const updates2 = {
```

Comment:
Should we rename this with a more meaningful name?

Reply:
done in https://github.com/pagopa/io-functions-commons/pull/160/commits/2af7d800afeddd5a07e269158cbe1e0e09d024d6